server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -681,6 +687,17 @@ public void addRemoteStoreCustomMetadata(IndexMetadata.Builder tmpImdBuilder, bo
         tmpImdBuilder.putCustom(IndexMetadata.REMOTE_STORE_CUSTOM_KEY, remoteCustomData);
     }
 
+    public ClusterState applyCreateIndexRequestWithNoTemplates(
+        final ClusterState currentState,
+        final CreateIndexClusterStateUpdateRequest request,
+        final boolean silent,
+        final BiConsumer<Metadata.Builder, IndexMetadata> metadataTransformer
+    ) throws Exception {
+        // Using applyCreateIndexRequestWithV1Templates with empty list instead of applyCreateIndexRequestWithV2Template
+        // with null template as applyCreateIndexRequestWithV2Template has assertions when template is null
+        return applyCreateIndexRequestWithV1Templates(currentState, request, silent, Collections.emptyList(), metadataTransformer);
```

Comment:
Is there any opportunity to abstract the common portions of `applyCreateIndexRequestWithV1Templates` and `applyCreateIndexRequestWithV2Templates`? I'm not very familiar with this file and trying to understand it in greater depth.

Reply:
> Is there any opportunity to abstract the common portions of `applyCreateIndexRequestWithV1Templates` and `applyCreateIndexRequestWithV2Templates`? 

There are actually 4 `applyCreateIndexRequest...()` methods and there are some commonalities.  However,

> I'm not very familiar with this file and trying to understand it in greater depth.

I doubt any of us area, as this is pre-fork code.  I'm probably the one who has spent the most time digging into it.  I thought about a refactoring as part of this, and concluded it would be much harder for anyone to review than the simple fix here.  In addition, I'm not sure much (if any) of this is tested.

Reply:
TLDR: @cwperks your suggestion is valid. Whoever refactors writes a bunch of tests.  You up for it? :)

Reply:
When I have some time to look into this file further, yes. I have wanted to take a deeper dive into mappings and templates at some point so this may be a good opportunity to do so.

Reply:
Marking this thread as resolved.